insonmnia/marketplace/server.go

```diff
@@ -0,0 +1,270 @@
+package marketplace
+
+import (
+	"errors"
+	"net"
+	"sync"
+
+	"github.com/pborman/uuid"
+	pb "github.com/sonm-io/core/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var (
+	errOrderNotFound     = errors.New("Order cannot be found")
+	errPriceIsZero       = errors.New("Order price cannot be less or equal than zero")
+	errOrderIsNil        = errors.New("Order cannot be nil")
+	errSlotIsNil         = errors.New("Order slot cannot be nil")
+	errResourcesIsNil    = errors.New("Slot resources cannot be nil")
+	errStartTimeAfterEnd = errors.New("Start time is after end time")
+	errStartTimeRequired = errors.New("Start time is required")
+	errEndTimeRequired   = errors.New("End time is required")
+)
+
+type OrderStorage interface {
+	GetOrders(slot *pb.Slot) ([]*pb.Order, error)
```

Comment:
Great idea: make `Order` wrapper, which is constructed from `pb.Order` and performs all required checks (like required fields). Otherwise all other implementation will suffer from doing the same things.

Protobuf sucks, yeah.

Make in a separate PR.